src/index.c

```diff
@@ -696,11 +711,20 @@ static void II_TesterFree(struct IndexCriteriaTester *ct) {
 static IndexCriteriaTester *II_GetCriteriaTester(void *ctx) {
   IntersectIterator *ic = ctx;
   IICriteriaTester *ict = rm_malloc(sizeof(*ict));
-  for (size_t i = 0; i < array_len(ic->its); ++i) {
-    ic->testers = array_append(ic->testers, IITER_GET_CRITERIA_TESTER(ic->its[i]));
+  for (size_t i = 0; i < ic->num; ++i) {
+    IndexCriteriaTester *tester = IITER_GET_CRITERIA_TESTER(ic->its[i]);
+    if (!tester) {
+      for (int j = 0; j < array_len(ic->testers); j++) {
+        ic->testers[i]->Free(ic->testers[i]);
+      }
+      array_free(ic->testers);
+      rm_free(ict);
```

Comment:
allocate ict after the array; this way there's one less free if not all children have cts.